receiver/cloudflarereceiver/config.go

```diff
@@ -39,8 +38,16 @@ func (c *Config) Validate() error {
 		return errNoEndpoint
 	}
 
-	if c.Logs.TLS == nil {
-		return errNoTLS
+	if c.Logs.TLS != nil {
+		// When TLS cert is specified, but nothing for key
+		if c.Logs.TLS.CertFile != "" && c.Logs.TLS.KeyFile == "" {
+			return errNoKey
+		}
+
+		// When TLS key is specified, but nothing for cert
+		if c.Logs.TLS.CertFile == "" && c.Logs.TLS.KeyFile != "" {
+			return errNoCert
+		}
```

Comment:
What about the case when `tls` is specified but neither key nor cert is? For what it's worth, I think it'd be fine to just check each field independently of the other.

I also think it's best if we aggregate these errors with any others which may occur, so the user can see multiple points of feedback at once.
```suggestion
		// When TLS cert is specified, but nothing for key
		if c.Logs.TLS.KeyFile == "" {
			errs = multierr.Append(errs, errNoKey)
		}

		// When TLS key is specified, but nothing for cert
		if c.Logs.TLS.CertFile == "" {
			errs = multierr.Append(errs, errNoCert)
		}
```

Reply:
Thanks for the suggestion. I committed this, but see there are some errors around errs being undefined. I believe I just need to move `var errs error` above. Testing that out now and will commit shortly.